### ISSUE TITLE
Support DateOnly and TimeOnly types

### ISSUE THREAD

**Issue by nhirschey** [fsprojects/FSharp.Data#1461]
When this is moved to >= .NET6.0, it would be nice to add support for System.DateOnly and System.TimeOnly types. DateOnly in particular, because often a date is just a date.

Mentioning now b/c I see the repackaging major version bump, though maybe requiring net6.0 is not possible because type providers have to stay netstandard2.0.

I could contribute a PR for the functionality when it’s allowed.

**Comment by cartermp** [comment]
Yeah, if we broke compat with .NET Framework then we could do this I think. But that's a pretty big change to make, and although the majority of F# developers are not using .NET Framework, there are enough to make breaking that existing compat probably not worth it for a while. FWIW if there were no compat with .NET Framework to speak of today, I'd be completely in favor of this and _not_ trying to find a way to add that compat.

**Comment by nhirschey** [comment]
Makes sense. Backwards compatibility is a great F# feature, and DateOnly is still young.

thanks for considering.
